### PR TITLE
Responsive Hamburger Menu

### LINKED ISSUE #144: Burger icon for responsive menu
Currently the menu has no responsiveness other than wrapping itself to fit the screen. 
We should add a burger menu to hold all menu items as it is standard for responsive apps.

The only concern I have is with how to deal with the modal for extensions in this case.

### PR DESCRIPTION
**Pull request by jorgeamadosoria**
Cutoff for the hamburger menu is 768px. Could be smaller but the amount and size of the icons is too much at 768px to keep in one line. 

The cutoff could be moved to 648 or smaller if necessary, if the icon sizes are changed for 768px. I like it as it is, but I'm not married to it.

A new menu.js file is introduced, to initialize the burger icon click. I think it's overkill, but that's the architecture, nothing to do about it.

The stylesheet is refactored to contain all the classes for a particular size together, order by size in descendent order. That way smaller resolution screens will overwrite bigger ones, as expected, and all classes for a particular resolution are lumped together. 

A new media query is added, with min-width: 768px. This media query contains the hover and focus effects for the icons in the menu, which are not needed in the responsive menu since there are no icons there.

Several new CSS classes are introduced as well as new ids for menu elements.

### CODE REVIEW

File: public/index.pug [hunk 1]
```diff
@@ -53,7 +53,16 @@ html(lang='en-US')
                   target='_blank'
                 ) #{ ext.nameAuthor }
     header.header
-      ul.header__list
+      .header__title-container
+        h1.header__title Simple Icons
+        p.header__description #{ iconCount } Free #[abbr(title='Scalable Vector Graphic') SVG] icons for popular brands
+      #menuToggle
```
Comment (mondeja) [inline public/index.pug:59]: We are not using camelCase right?

```suggestion
      #menu-toggle
```

File: public/index.pug [hunk 2]
```diff
@@ -298,7 +313,6 @@ html(lang='en-US')
           src='//cdn.carbonads.com/carbon.js?serve=CKYIPK7M&placement=simpleiconsorg',
           type='text/javascript'
         )
-
```
Comment (mondeja) [inline public/index.pug]: Seems to me that this line is fine as it is.

File: public/stylesheet.css [hunk 3]
```diff
@@ -1,3 +1,32 @@
...
+  color: var(--color-text-default);
+}
+.header__list svg {
+  display: block;
+}
+input#hamburger {
+  display: none;
+}
+
+#menuToggle {
```
Comment (mondeja) [inline public/stylesheet.css:20]: Here also

```suggestion
#menu-toggle {
```

File: public/stylesheet.css [hunk 4]
```diff
@@ -931,3 +847,288 @@ body.order-by-relevance .grid-item {
...
+    font-style: normal;
+    font-size: 2em;
+    padding: 10px;
+  }
+
+  .header input:checked ~ ul.header__list {
+    display: flex;
+  }
+
+  #menuToggle {
```
Comment (mondeja) [inline public/stylesheet.css:941]:
```suggestion
  #menu-toggle {
```

File: public/stylesheet.css [hunk 5]
```diff
@@ -931,3 +847,288 @@ body.order-by-relevance .grid-item {
...
+    display: block;
+    position: relative;
+
+    -webkit-user-select: none;
+    user-select: none;
+  }
+
+  #menuToggle a {
```
Comment (mondeja) [inline public/stylesheet.css:949]:
```suggestion
  #menu-toggle a {
```

File: public/stylesheet.css [hunk 6]
```diff
@@ -931,3 +847,288 @@ body.order-by-relevance .grid-item {
...
+    text-decoration: none;
+    color: #232323;
+
+    transition: color 0.3s ease;
+  }
+
+  #menuToggle input {
```
Comment (mondeja) [inline public/stylesheet.css:956]:
```suggestion
  #menu-toggle input {
```

File: public/stylesheet.css [hunk 7]
```diff
@@ -931,3 +847,288 @@ body.order-by-relevance .grid-item {
...
+    opacity: 0; /* hide this */
+    z-index: 2; /* and place it over the hamburger */
+
+    -webkit-touch-callout: none;
+  }
+
+  /*
+ * Just a quick hamburger
+ */
+  #menuToggle span {
```
Comment (mondeja) [inline public/stylesheet.css:975]:
```suggestion
  #menu-toggle span {
```

File: public/stylesheet.css [hunk 8]
```diff
@@ -931,3 +847,288 @@ body.order-by-relevance .grid-item {
...
+
+  #menuToggle span:nth-last-child(2) {
+    transform-origin: 0% 100%;
+  }
+
+  /* 
+ * Transform all the slices of hamburger
+ * into a crossmark.
+ */
+  #menuToggle input:checked ~ span {
```
Comment (mondeja) [inline public/stylesheet.css:1002]:
```suggestion
  #menu-toggle input:checked ~ span {
```

File: public/stylesheet.css [hunk 9]
```diff
@@ -931,3 +847,288 @@ body.order-by-relevance .grid-item {
...
+    opacity: 1;
+    transform: rotate(45deg) translate(-2px, -1px);
+    background: #232323;
+  }
+
+  /*
+ * But let's hide the middle one.
+ */
+  #menuToggle input:checked ~ span:nth-last-child(3) {
```
Comment (mondeja) [inline public/stylesheet.css:1011]:
```suggestion
  #menu-toggle input:checked ~ span:nth-last-child(3) {
```

File: public/stylesheet.css [hunk 10]
```diff
@@ -931,3 +847,288 @@ body.order-by-relevance .grid-item {
...
+    opacity: 0;
+    transform: rotate(0deg) scale(0.2, 0.2);
+  }
+
+  /*
+ * Ohyeah and the last one should go the other direction
+ */
+  #menuToggle input:checked ~ span:nth-last-child(2) {
```
Comment (mondeja) [inline public/stylesheet.css:1019]:
```suggestion
  #menu-toggle input:checked ~ span:nth-last-child(2) {
```

File: public/stylesheet.css [hunk 11]
```diff
@@ -931,3 +847,288 @@ body.order-by-relevance .grid-item {
...
+  }
+
+  #menu li {
+    font-size: 22px;
+  }
+
+  /*
+ * And let's slide it in from the left
+ */
+  #menuToggle input:checked ~ ul {
```
Comment (mondeja) [inline public/stylesheet.css:1047]:
```suggestion
  #menu-toggle input:checked ~ ul {
```

File: public/stylesheet.css [hunk 12]
```diff
@@ -931,3 +847,288 @@ body.order-by-relevance .grid-item {
...
+    margin: 0.5rem 0 0.5rem 1rem;
+  }
+
+  .control-field:first-of-type {
+    flex-basis: 70%;
+    flex-grow: 1;
+    margin: 0.5rem 0;
+  }
+
+  #menuToggle {
```
Comment (mondeja) [inline public/stylesheet.css:1077]:
```suggestion
  #menu-toggle {
```

File: public/stylesheet.css [hunk 13]
```diff
@@ -931,3 +847,288 @@ body.order-by-relevance .grid-item {
...
+    display: block;
+
+    -webkit-user-select: none;
+    user-select: none;
+  }
+
+  #menuToggle input {
```
Comment (mondeja) [inline public/stylesheet.css:1084]:
```suggestion
  #menu-toggle input {
```

File: public/stylesheet.css [hunk 14]
```diff
@@ -931,3 +847,288 @@ body.order-by-relevance .grid-item {
...
+    left: 0;
+    background: #cdcdcd;
+    border-radius: 3px;
+
+    z-index: 15;
+
+    transform-origin: 4px 0px;
+  }
+
+  #menuToggle span:first-child {
```
Comment (mondeja) [inline public/stylesheet.css:990]:
```suggestion
  #menu-toggle span:first-child {
```

File: public/stylesheet.css [hunk 15]
```diff
@@ -931,3 +847,288 @@ body.order-by-relevance .grid-item {
...
+    transform-origin: 0% 0%;
+  }
+
+  #menuToggle span:nth-last-child(2) {
```
Comment (mondeja) [inline public/stylesheet.css:994]:
```suggestion
  #menu-toggle span:nth-last-child(2) {
```

File: public/scripts/menu.js [hunk 16]
```diff
@@ -0,0 +1,7 @@
+export default async function initMenu(document, domUtils) {
+  document
+    .querySelector('#menuToggle svg')
```
Comment (mondeja) [inline public/scripts/menu.js:3]:
```suggestion
    .querySelector('#menu-toggle svg')
```

File: public/scripts/menu.js [hunk 17]
```diff
@@ -0,0 +1,7 @@
...
+    .addEventListener('click', (e) =>
+      domUtils.toggleClass(document.querySelector('#menu'), 'hidden-768'),
```
Comment (mondeja) [inline public/scripts/menu.js]: Why this strange class name `hidden-768`?

**Review by mondeja — DISMISSED** [review]

File: public/scripts/menu.js (hunk 17 above)
Comment (jorgeamadosoria) [reply public/scripts/menu.js]: it's hidden in >768px.
there is a hidden class already, so this is descriptive.

Comment (mondeja) [reply public/scripts/menu.js]: Maybe would be better to name it as `hidden-gt-768px`?

Comment (jorgeamadosoria) [reply public/scripts/menu.js]: sure, if you think that's best. Done.

File: public/index.pug [hunk 18]
```diff
@@ -53,7 +53,16 @@ html(lang='en-US')
...
-      ul.header__list
+      .header__title-container
+        h1.header__title Simple Icons
+        p.header__description #{ iconCount } Free #[abbr(title='Scalable Vector Graphic') SVG] icons for popular brands
+      #menu-Toggle
+        svg(viewBox='0 0 100 80', width='40', height='40')
+          rect(width='100', height='20')
+          rect(y='30', width='100', height='20')
+          rect(y='60', width='100', height='20')
+        input.hidden(type='checkbox')
```
Comment (mondeja) [inline public/index.pug:64]: What is doing this input? It is not needed, I think 🤔

File: public/index.pug [hunk 19]
```diff
@@ -53,7 +53,16 @@ html(lang='en-US')
                   target='_blank'
                 ) #{ ext.nameAuthor }
     header.header
-      ul.header__list
+      .header__title-container
+        h1.header__title Simple Icons
+        p.header__description #{ iconCount } Free #[abbr(title='Scalable Vector Graphic') SVG] icons for popular brands
+      #menu-Toggle
+        svg(viewBox='0 0 100 80', width='40', height='40')
```
Comment (mondeja) [inline public/index.pug:60]: We reallly need a wrapper container for the SVG? I think that the SVG can be the container itself. Also, the identifier breaks our style, it shouldn't be capitalized: `menu-toggle`.

File: public/scripts/menu.js [hunk 20]
```diff
@@ -0,0 +1,7 @@
+export default async function initMenu(document, domUtils) {
+  document
+    .querySelector('#menu-Toggle svg')
+    .addEventListener('click', (e) =>
+      domUtils.toggleClass(document.querySelector('#menu'), 'hidden-gt-768px'),
```
Comment (mondeja) [inline public/scripts/menu.js:5]: The event is not needed here.

```diff
-    .addEventListener('click', (e) =>
+    .addEventListener('click', () =>
```

File: public/stylesheet.css [hunk 21]
```diff
@@ -931,3 +847,288 @@ body.order-by-relevance .grid-item {
...
+    opacity: 1;
+    text-decoration: underline;
+  }
+
+  .copy-button:not(:disabled):hover::before {
+    cursor: pointer;
+    display: block;
+    z-index: 1;
+  }
+}
```
Comment (mondeja) [inline public/stylesheet.css:862]: Why do you've moved all the `@media` to the end? I think that this confuses more than clarifies.

File: public/stylesheet.css [hunk 22]
```diff
@@ -931,3 +847,288 @@ body.order-by-relevance .grid-item {
...
+    top: -7px;
+    left: -5px;
+
+    cursor: pointer;
+
+    opacity: 0; /* hide this */
+    z-index: 2; /* and place it over the hamburger */
+
+    -webkit-touch-callout: none;
+  }
```
Comment (mondeja) [inline public/stylesheet.css:970]: As before, I don't know that is this.

File: public/stylesheet.css [hunk 23]
```diff
@@ -931,3 +847,288 @@ body.order-by-relevance .grid-item {
...
+    opacity: 0;
+    transform: rotate(0deg) scale(0.2, 0.2);
+  }
+
+  /*
+ * Ohyeah and the last one should go the other direction
+ */
+  #menu-Toggle input:checked ~ span:nth-last-child(2) {
+    transform: rotate(-45deg) translate(0, -1px);
+  }
```
Comment (mondeja) [inline public/stylesheet.css:1021]: This is doing nothing.

**Review by mondeja — CHANGES_REQUESTED** [review]
Hi @jorgeamadosoria. Are you still interested in this PR? I guess that the functionality relative to the input is currently broken.

**Comment by mondeja** [comment]
I'm closing this due to inactivity and very deep conflicts between branches.

**Pull request closed without merging**